index.html

```diff
@@ -101,212 +220,1289 @@ <h3>
+              data: encoder.encode("PING")
+            };
+
+            await writer.write(message);
+
+            writer.releaseLock();
           </pre>
-          <p>
-            The <dfn>UDPSocketOptions</dfn> dictionary inherits all attributes 
-            from {{SocketOptions}}.
-          </p>
-        </section>
-        <section data-dfn-for="TCPSocket">
-          <h3>
-            `TCPSocket` interface
-          </h3>
-          <pre class="idl">
-          [Exposed=Window, SecureContext] interface TCPSocket {
-            Promise&lt;undefined&gt; close();
-
-            readonly attribute DOMString remoteAddress;
-            readonly attribute unsigned short remotePort;
-            readonly attribute ReadableStream readable;
-            readonly attribute WritableStream writable;
-          };
+
+          The {{WritableStreamDefaultWriter/write()}} method returns a {{Promise}} which
+          resolves when data has been written. While having some data available in
+          the transmit buffer is important to maintain good throughput awaiting this
+          {{Promise}} before generating too many chunks of data is a good practice
+          to avoid excessive buffering.
+        </aside>
+
+        The steps to <dfn data-lt="initialize udp writable stream">initialize the UDP writable stream</dfn> are:
+
+        <ol>
+          <li>
+            Let |stream:WritableStream| be a [=new=] {{WritableStream}}.
+          <li>
+            Let |signal:AbortSignal| be |stream|'s [=WritableStream/signal=].
+          <li>
+            Let |writeAlgorithm| be the following steps, given |chunk|:
+            <ol>
+              <li>
+                Let |promise:Promise| be [=a new promise=].
+              <li>
+                Assert: |signal| is not [=AbortSignal/aborted=].
+              <li>
+                Let |message:UDPMessage| be a new {{UDPMessage}}.
+              <li>
+                If |chunk| cannot be [=converted to an IDL value=] of type
+                {{UDPMessage}}, reject |promise| with a {{TypeError}} and return
+                |promise|. Otherwise, save the result of the conversion in
+                |message|.
+              <li>
+                If either |message|["{{UDPMessage/remoteAddress}}"]
+                or |message|["{{UDPMessage/remotePort}}"] is specified,
+                reject |promise| with a {{TypeError}} and return |promise|.
+                <div class="note">
+                  The |message|["{{UDPMessage/remoteAddress}}"]
+                  and |message|["{{UDPMessage/remotePort}}"]
+                  currently have no effect on sending packets since
+                  the UDP socket is only allowed to operate in the connected mode.
+                </div>
+              <li>
+                [=In parallel=], run the following steps:
+                <ol>
+                  <li>
+                    Invoke the operating system to send |message|["{{UDPMessage/data}}"] to the socket.
```

Comment:
Same here. Need to get a copy of the payload as a byte sequence before beginning the "in parallel" steps.

Reply:
Fixed.